csharp/ql/test/library-tests/controlflow/graph/ExitElement.expected

```diff
@@ -307,7 +307,7 @@
 | ArrayCreation.cs:9:43:9:50 | { ..., ... } | ArrayCreation.cs:9:43:9:50 | { ..., ... } | normal |
 | ArrayCreation.cs:9:45:9:45 | 2 | ArrayCreation.cs:9:45:9:45 | 2 | normal |
 | ArrayCreation.cs:9:48:9:48 | 3 | ArrayCreation.cs:9:48:9:48 | 3 | normal |
-| Assert.cs:8:5:12:5 | {...} | Assert.cs:10:9:10:31 | call to method Assert | exit |
```

Comment:
@hvitved These changes seem to be `Debug.Assert` calls, which overlap between `SystemDiagnosticsCodeAnalysisDoesNotReturnIfAnnotatedAssertTrueMethod` and `SystemDiagnosticsDebugAssertTrueMethod`. In a general case the attribute behaviour ("does not return") will be implemented as an exception throwing, but in case of `Debug.Assert` it's a message box, and `System.Environment.Exit(1)`. 

One way of handling this conflict would be to exclude debug assert from `SystemDiagnosticsCodeAnalysisDoesNotReturnIfAnnotatedAssertTrueMethod`. What do you think?

On a related note, `AssertMethod`s currently don't have a way to specify that they throw an exception or they are `ExitingCall`s. Currently these options seem to be exclusive. We could extend the classes to support this, but it's probably not worth the effort.